e2e/testcafe-devextreme/tests/dataGrid/scrolling.ts

```diff
@@ -1720,3 +1720,47 @@ test('The row alternation should display correctly when grouping and virtual scr
   grouping: { autoExpandAll: true },
   scrolling: { mode: 'virtual', useNative: false },
 })));
+
+test('DataGrid - Gray boxes appear when the push method is used to remove rows in infinite scrolling mode (T1240079)', async (t) => {
+  const dataGrid = new DataGrid('#container');
+  const { takeScreenshot, compareResults } = createScreenshotsComparer(t);
+  await t
+    .click(dataGrid.getDataCell(0, 0).element)
```

Comment:
Let's use .apiPush() method instead of clicking for deleting items

```suggestion
  await dataGrid.apiPush([
    { type: 'remove', key: 1 },
    { type: 'remove', key: 2 },
  ]);
```